sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -321,6 +333,20 @@ case class StaticInvoke(
     copy(arguments = newChildren)
 }
 
+object StaticInvoke {
+  def apply(
```

Comment:
As we have second constructor, do we need `apply`?

Reply:
The `this` constructor ends up being a static method on `StaticInvoke`, the `apply` as a method on `StaticInvoke$.class` which serves as the companion object of `StaticInvoke`. They are both needed for binary compatibility. The Scala compile generates both of them for the default constructor.